compiler/rustc_middle/src/infer/canonical.rs

```diff
@@ -286,10 +286,22 @@ impl<'tcx, V> Canonical<'tcx, V> {
 pub type QueryOutlivesConstraint<'tcx> =
     ty::Binder<ty::OutlivesPredicate<GenericArg<'tcx>, Region<'tcx>>>;
 
+impl<'tcx, V: 'tcx> ty::Lift<'tcx> for Canonical<'tcx, V>
```

Comment:
I was forced to implement `Lift` for `Canonical` and `CanonicalUserTypeAnnotation` manually, but I'll admit I'm not sure why the normal `derive` was insufficient.